src/wasm-binary.h

```diff
@@ -774,7 +775,7 @@ class WasmBinaryBuilder {
 
   size_t pos = 0;
   Index startIndex = -1;
-  bool useDebugLocation;
+  std::pair<Function::DebugLocation, bool> debugLocation;
```

Comment:
not obvious what the bool means

Reply:
since std::optional is available in C++17, I choose to do its replacement std::pair<T, bool> -- I can do with some custom type if you want.

Reply:
If it's to check existance, can we use an `std::set` of the items, and check if it's there etc.?